django/views/debug.py

```diff
@@ -544,6 +545,13 @@ def get_exception_traceback_frames(self, exc_value, tb):
                 pre_context = []
                 context_line = "<source code not available>"
                 post_context = []
+            spaces = "\n" + " " * (
+                len(context_line)
+                - len(context_line.lstrip())
+                + len(str(lineno + 1))
+                - 2
+            )
+            colno = spaces + traceback.format_tb(tb, limit=1)[0].split("\n")[-2]
```

Comment:
I'd prefer to use `tb.tb_frame.f_code.co_positions()` instead of cutting the traceback on Python 3.11+ .

We need to take into account that `start_column` and `end_column` are `None` when `-X no_debug_ranges` is used.

Reply:
Yes please use the API rather than parsing the text that Python happens to output right now

Reply:
Thanks for the feedback! Agreed, I'd rather use the API, but had some difficulties.
If I access `co_positions()` there are way to many tuples being returned, how do I know which one is the one I need ?

Reply:
I don't know - you'll need to read the feature description and PEP as linked in ticket: https://docs.python.org/3.11/whatsnew/3.11.html#pep-657-fine-grained-error-locations-in-tracebacks

Reply:
Yes, I've read through them several times, but it's still not completely clear to me. The "reference implementation" link is also broken. So any help is greatly appreciated!

Reply:
The relevant bits from the `co_positions` docs are:

> The iterator returns tuples containing the (start_line, end_line, start_column, end_column). The i-th tuple corresponds to the position of the source code that compiled to the i-th instruction. Column information is 0-indexed utf-8 byte offsets on the given source line.

And 

> ...
> When this occurs, some or all of the tuple elements can be None.

Reply:
@ammaraskar Thanks for your reply! This is the part that I get. But after getting these disassembled instructions, how do I know in which one the error occured ? For any given function potentially hundreds of tuples/instructions will be returned.

Reply:
Aah, that is given by `tb.tb_lasti`. See this code in the traceback module: https://github.com/python/cpython/blob/f09da28768b77713566e932e912f107b6b57e8fd/Lib/traceback.py#L354

The higher level api to get this info would be to do `StackSummary.extract()` with whatever `limit` you want and then you can iterate over the summary to get `FrameSummary` objects which have a `frame_summary.colno` and `frame_summary.end_colno` fields.

Reply:
Thanks so much! `tb.tb_lasti` is probably what I was missing. I'll get right back to work and update the PR in the next days!